docs/guides/intro-to-tests.md

```diff
@@ -0,0 +1,154 @@
+# Introduction to tests
+
+## What does this cover
+
+This introduction is designed to give you a high-level overview on the basics of testing, which you'll need when you're tackling the problems in your coursework at Code Your Future.
+In this introduction to testing we'll cover the following things:
+
+- Code specification
+- What is a test ?
+- Anatomy of a test ( using Jest )
+- Understanding test feedback ( using Jest )
+
+## Code specification
+
+`createWordCount` 🧮
+
+Let's consider a function called `createWordCount` that needs to work as follows:
+
+- it takes one argument - a _string_ of words separated by spaces, e.g. `"hello world"`
+- returns an _object_ with a count of each word in the string
```

Comment:
We're using this for JS1 week 1, so objects aren't something our trainees have encountered yet - maybe let's rework this example to just return a count of words (i.e. `createWordCount("hello world") === 2`)?

There are still plenty of edge-cases we can bring up (e.g. empty string, do numbers count as words, hyphenated words). Speaking of edge-cases, can we explicitly call these out? Something like:

> When thinking about a specification, and what tests you'll need, you should be thinking _both_ about how people would _normally_ use the function, and about _unusual_ examples that may be confusing or surprising (often called "edge cases"). For example, with our `createWordCount` example, what should happen when the empty string is used as an argument? What should happen for hyphenated words (e.g. North-West)? Do numbers count as words? What should `createWordCount("There were 17 trainees in the North-West class")` return?
>
> You should make sure you think about these edge-cases when you're approaching a problem, to make sure you solve it completely.
>
> Often when you're given a problem, not all of these edge-cases are described for you, so you'll need to use your best judgement (this is true as a professional developer too!). Writing down what you decided (either in a specification, or as a test) can be useful to show other developers what you decided. It will help them to understand your code, and also means that if they have different ideas about some of the cases, you can have a discussion about it. So in this example, we may write a test showing that numbers are counted as words (or that numbers are not counted as words, depending on what we decided!).

Reply:
I agree - it did occur to me that they won't have seen objects but I ploughed on anyway. I think the simplified version will work a lot better.